src/core/jit/synth/synth.rs

```diff
@@ -322,8 +322,13 @@ mod tests {
         let config = ConfigModule::from(config);
 
         let builder = Builder::new(&Blueprint::try_from(&config).unwrap(), doc);
-        let plan = builder.build(&Variables::new(), None).unwrap();
-        let plan = plan.try_map(Deserialize::deserialize).unwrap();
+        let plan = builder.build(None).unwrap();
+        let plan = plan
+            .try_map(|v| {
+                let serde = v.into_json().unwrap();
+                Deserialize::deserialize(serde)
+            })
+            .unwrap();
```

Comment:
The changed code converts first to `serde_json::Value` then deserializes it to the expected value instead of the previous implementation that deserializes straight to the output.
It's not that important for the test but raised an important question about optimization with `serde_json_borrow`. Since in the example we create a value and then drop it immediately it seems like the output value isn't tied with input in any way therefore no benefits of `serde_json_borrow`. And since we are using async_graphql that's already parsed the input to owned strings we can just pass the ownership of those strings anyway (async_graphql parsed everything to `async_graphql_value::Value` so we already did a lof of allocations and copying). For IR it could be still beneficial though

Reply:
I got your point but I am still not sure what you want to convey with this comment..